Poul12/SimpleToDoApp/src/main/java/pl/_1024kb/stowarzyszenienaukijavy/simpletodo/model/service/TaskServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+package pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.api.TaskDao;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.api.TaskService;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.daojdbc.DaoFactory;
+import pl._1024kb.stowarzyszenienaukijavy.simpletodo.model.entity.Task;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TaskServiceImpl implements TaskService
+{
+    private static TaskServiceImpl instance;
+    private DaoFactory factory = DaoFactory.getDaoFactory(DaoFactory.MYSQL_DAO);
+    private TaskDao dao = factory.getTaskDao();
+    //private MysqlTaskDao jdbcDao = new MysqlTaskDao();
+    private UserServiceImpl userService = UserServiceImpl.getInstance();
+    private static final Logger logger = LoggerFactory.getLogger(TaskServiceImpl.class);
+
+    private TaskServiceImpl(){
+        if(instance != null)
+            throw new IllegalStateException("Cannot create new instance, use getInstance");
+    }
+
+    public static TaskServiceImpl getInstance()
+    {
+        if(instance == null)
+                instance = new TaskServiceImpl();
+
+        return instance;
+    }
+
+    @Override
+    public String createTask(Task task, String username)
+    {
+        String messageInfo = "Pomyślnie zapisano zadanie ;)";
+        Long userId = 0L;
+
+        if(userService.getUserByUsername(username).isPresent())
```

Comment:
1. Klamry
2. Złe użycie Optionala - jeśli używasz isPreset oraz get to równie dobrze możesz sprawdzać czy jest różny od nulla. Optional ma takie fajne metody jak orElse lub orElseThrow.